src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Divide.cs

```diff
@@ -70,11 +72,80 @@ public static void Divide<T>(T x, ReadOnlySpan<T> y, Span<T> destination)
         internal readonly struct DivideOperator<T> : IBinaryOperator<T> where T : IDivisionOperators<T, T, T>
         {
             public static bool Vectorizable => typeof(T) == typeof(float)
-                                            || typeof(T) == typeof(double);
+                                            || typeof(T) == typeof(double)
+#if NET10_0_OR_GREATER
+                                            || ((Avx512F.IsSupported || Avx.IsSupported) && typeof(T) == typeof(int));
+#else
+                ;
+#endif
             public static T Invoke(T x, T y) => x / y;
-            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y) => x / y;
-            public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> y) => x / y;
-            public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y) => x / y;
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
+            {
+#if NET10_0_OR_GREATER
+                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+                {
+                    return x / y;
+                }
+
+                Debug.Assert(Avx.IsSupported && typeof(T) == typeof(int));
+                Vector128<int> denominator_zero = Sse2.CompareEqual(y.AsInt32(), Vector128<int>.Zero);
+                if (denominator_zero != Vector128<int>.Zero)
+                {
+                    throw new DivideByZeroException();
+                }
+
+                Vector256<double> num_pd = Avx.ConvertToVector256Double(x.AsInt32());
+                Vector256<double> den_pd = Avx.ConvertToVector256Double(y.AsInt32());
+                Vector256<double> div_pd = Avx.Divide(num_pd, den_pd);
+                Vector128<int> div_epi32 = Avx.ConvertToVector128Int32WithTruncation(div_pd);
+                return div_epi32.As<int, T>();
```

Comment:
I think it'd be better to use the xplat algorithm around `ConvertToDouble(WidenLower(x.AsInt32())) / ConvertToDouble(WidenLower(y.AsInt32()))` so it can lightup for .NET 8/9.

We should then ideally accelerate `Vector128.operator /` on .NET 10 and just use that there, so it can be directly handled and lightup anyone else using `operator /` as well.